src/routes/add-custom-discipline/components/QRCodeReaderOverlay.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { ReactElement, useEffect, useRef, useState } from 'react'
+import { AppState, Linking, Modal, PermissionsAndroid } from 'react-native'
+import { BarCodeReadEvent, RNCamera } from 'react-native-camera'
+import styled from 'styled-components/native'
+
+import { CloseCircleIconBlue, CloseCircleIconWhite } from '../../../../assets/images'
+import Button from '../../../components/Button'
+import { BUTTONS_THEME } from '../../../constants/data'
+import labels from '../../../constants/labels.json'
+
+const Container = styled.View`
+  flex: 1;
+  background-color: ${props => props.theme.colors.white};
+`
+
+const Icon = styled.TouchableOpacity`
+  position: absolute;
+  top: 10px;
+  right: 10px;
+  width: 30px;
+  height: 30px;
+`
+
+const Camera = styled(RNCamera)`
+  flex: 1;
+  position: relative;
+  margin: 50px 0 0;
+`
+
+const NotAuthorisedView = styled.View`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 100%;
+  background-color: ${props => props.theme.colors.lunesWhite};
+`
+
+const NoAuthDescription = styled.Text`
+  font-family: ${props => props.theme.fonts.contentFontRegular};
+  font-size: ${props => props.theme.fonts.defaultFontSize};
+  color: ${props => props.theme.colors.lunesGreyMedium};
+  padding-bottom: 20px;
+`
+
+interface Props {
+  setVisible: (input: boolean) => void
+  setCode: (input: string) => void
+}
+
+const AddCustomDisciplineScreen = ({ setVisible, setCode }: Props): ReactElement => {
+  const appState = useRef(AppState.currentState)
+
+  const [isPressed, setIsPressed] = useState<boolean>(false)
+  const [requestPermissions, setRequestPermissions] = useState<boolean>(false)
+  const [permissionDenied, setPermissionDenied] = useState<boolean>(false)
+
+  const onBarCodeRead = (scanResult: BarCodeReadEvent) => {
+    setCode(scanResult.data)
+    setVisible(false)
+  }
+
+  useEffect(() => {
+    if (requestPermissions) {
```

Comment:
Could you explain this? AFAICS `requestPermissions` is only ever set to `true` if the app comes from background to foreground while this is rendered. So if I open the app and navigate here without putting it in the background, the permission is never requested? What am I missing here? :see_no_evil:

Reply:
Yeah maybe i should add a comment here.
This is only needed when you got to the page, deny camera permissions, click then on settings. When then navigating back from settings this code is executed, leading to a second request for permission. The first one is triggered automatically by RNCamera, therefore no need for us to handle.

Not sure if this is the best solution, but i couldn't work out a better on. There is a `onStatusChange` function of RNCamera, but it does not get called (correctly), i guess because there is nor rerender, when navigating back from settings. I tried a lot of different approaches, but either i got stuck in a infinite rerender cycle or the sencond prompt for permissions was not fired. So I ended up with this solution.

Reply:
oke maybe this is important to mention: it is only needed when navigate to settings and then select "ask every time", when selecting "allow while using the app" it would work fine without this.

Reply:
Can't we just always request permissions in a useFocusEffect/app state change? Doesn't hurt to request permissions twice, right?

Reply:
I could not get this working, always leading to a loop, if you can make a working solution with useFocusEffect or anything, I would be happy to see it. For me it always leaded into an endless loop of active / focused and opening and closing the request-dialog.

Reply:
will try and let you know :)

Reply:
oke more or less just renaming. Thought it would be cool to get rid of the useEffect or listening to the AppState.